File: src/main/java/tri/le/music/entity/SongEntity.java
package tri.le.music.entity;

import javax.persistence.*;

/**
 * Created by TriLe on 3/5/2016.
 */
@Entity
public class SongEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String name;

    @ManyToOne
    private ArtistEntity artist;

    @ManyToOne
    private AlbumEntity album;

    private String composer;

    private String genre;

    public SongEntity() {
    }

    public SongEntity(String name) {
        this.name = name;
    }

    public SongEntity(String name, String genre) {
        this.name = name;
        this.genre = genre;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public ArtistEntity getArtist() {
        return artist;
    }

    public void setArtist(ArtistEntity artist) {
        this.artist = artist;
    }

    public AlbumEntity getAlbum() {
        return album;
    }

    public void setAlbum(AlbumEntity album) {
        this.album = album;
    }

    public String getComposer() {
        return composer;
    }

    public void setComposer(String composer) {
        this.composer = composer;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }
}

File: src/main/java/tri/le/music/entity/SongEntity.kt
package tri.le.music.entity

import javax.persistence.*

/**
 * Created by TriLe on 3/5/2016.
 */
@Entity
data class SongEntity(
        var name: String = "",

        var description: String = "",

        @ManyToOne
        var album: AlbumEntity? = null,

        @ManyToOne
        var artist: HumanEntity? = null,

        @ManyToOne
        var composer: HumanEntity? = null,

        var genres: String = ""
) {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long = 0

    constructor(name: String, album: AlbumEntity?) : this(name, "", album, null, null, "")

    constructor(name: String) : this(name, null)

    constructor() : this("")
}
